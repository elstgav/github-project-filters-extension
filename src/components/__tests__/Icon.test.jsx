import Icon from 'components/Icon'
import EnzymeTestWrapper from 'test/support/EnzymeTestWrapper'

const test = new EnzymeTestWrapper(Icon)

describe('Icon', () => {
  beforeEach(() => {
    test.props.icon = Object.keys(Icon.paths)[0] // Pick an existing icon for the test
  })

  it('displays a <svg> element', () => {
    expect(test.Icon.find('svg')).toHaveLength(1)
    expect(test.Icon.find('path').prop('d')).toBe(Icon.paths[test.props.icon])
    expect(test.Icon.props().className).toBe(`octicon octicon-${test.props.icon}`)
    expect(test.Icon).toMatchSnapshot()
  })

  it('has a configurable className', () => {
    test.props.className = 'foo'
    expect(test.Icon.props().className).toBe(`octicon octicon-${test.props.icon} foo`)
  })

  it('sets [aria-hidden] to true unless thereâ€™s an [aria-label]', () => {
    expect(test.Icon.prop('aria-hidden')).toBe(true)

    test.Icon.setProps({ ariaLabel: 'Foo' })

    expect(test.Icon.prop('aria-hidden')).toBe(false)
    expect(test.Icon.prop('aria-label')).toBe('Foo')
  })

  it('has configurable size', () => {
    test.props.height = 10
    test.props.width  = 10

    expect(test.Icon.props()).toMatchObject({ height: 10, width: 10 })
  })

  it('defaults to a square size if height is not set', () => {
    test.props.width = 42

    expect(test.Icon.props().height).toBe(42)
  })
})
