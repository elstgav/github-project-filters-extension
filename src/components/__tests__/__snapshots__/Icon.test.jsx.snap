// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon displays a <svg> element 1`] = `
<svg
  aria-hidden={true}
  aria-label=""
  className="octicon octicon-x"
  height={16}
  role="img"
  version="1.1"
  viewBox="0 0 12 16"
  width={16}
>
  <path
    d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"
    fillRule="evenodd"
  />
</svg>
`;
